src/quick-lint-js/error.h

```diff
@@ -237,6 +237,13 @@
       { source_code_span escape_sequence; },                                   \
       .error(QLJS_TRANSLATABLE("code point out of range"), escape_sequence))   \
                                                                                \
+  QLJS_ERROR_TYPE(                                                             \
+      error_escaped_code_point_in_unicode_out_of_range, "E207",                \
+      { source_code_span escape_sequence; },                                   \
+      .error(QLJS_TRANSLATABLE("Code point in unicode escape sequence must "   \
```

Comment:
Nit: Diagnostics start with a lower-case letter.

Nit: Unicode is always capitalized in documentation.

```suggestion
      .error(QLJS_TRANSLATABLE("code point in Unicode escape sequence must "   \
```